### PR TITLE
Update tests for probabilities and change how the economics config is loaded and passed.

### PR DESCRIPTION
**Pull request by ajlacey**
First, my apologies for the size of this PR, I can split it up if that's better for reviewing.

What this PR does:
updates the setup file to have `MustLoadEconomicsConfig` instead of `LoadEconomicsConfig`. This function is then used across handler init()'s and in the handler functions (more updates are needed here, but I think that should be for a future effort). The must function will either load the config or log a fatal error and close the application. This is was already being done across handlers, but now it is consolidated with the actual loading of the config.

Adds the ability to calculate a market probability at a point in time based on the starting conditions and the current yes/no tally.
Adds corresponding tests for calculating market probabilities

Moves tests for wpam market probabilities to the wpam package and adds the dependency injection needed to test this function.

### CODE REVIEW

File: backend/handlers/markets/marketdetailshandler.go [hunk 1]
```diff
@@ -51,14 +52,10 @@ func MarketDetailsHandler(w http.ResponseWriter, r *http.Request) {
...
 	// Calculate probabilities using the fetched bets
-	probabilityChanges := wpam.CalculateMarketProbabilitiesWPAM(publicResponseMarket.CreatedAt, bets)
+	probabilityChanges := wpam.CalculateMarketProbabilitiesWPAM(setup.MustLoadEconomicsConfig, publicResponseMarket.CreatedAt, bets)
 
 	// find the number of users on the market
 	numUsers := models.GetNumMarketUsers(bets)
-	if err != nil {
-		http.Error(w, "Error retrieving number of users.", http.StatusInternalServerError)
-		return
-	}
```
Comment (ajlacey) [inline backend/handlers/markets/marketdetailshandler.go:61]: GetNumMarketUsers does not return an error

File: backend/setup/setup.go [hunk 2]
```diff
@@ -3,54 +3,69 @@ package setup
...
+
+func mustLoadEconomicsConfig() {
+	economicConfig = &EconomicConfig{}
+	err := yaml.Unmarshal(setupYaml, economicConfig)
+	if err != nil {
+		log.Fatal("Error parsing YAML config:", err) // If the config cannot be loaded, the application cannot recover.
+	}
+}
+
+func init() {
```
Comment (ajlacey) [inline backend/setup/setup.go:69]: runs before main

File: backend/setup/setuptesting.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+		Economics: Economics{
+			MarketCreation: MarketCreation{
+				InitialMarketProbability:   probability,
+				InitialMarketSubsidization: subsidization,
+				InitialMarketYes:           yes,
+				InitialMarketNo:            no,
+			},
+		},
+	}
+}
```
Comment (ajlacey) [inline backend/setup/setuptesting.go:18]: Helper for tests that need part of/or an entire EconomicConfig

File: backend/tests/dbpm/1_divideupmarketpoolsharesdbpm_test.go [hunk 4]
```diff
@@ -2,14 +2,15 @@ package test
...
+	"socialpredict/setup"
 	test "socialpredict/tests"
 	"testing"
 )
 
 func TestDivideUpMarketPoolSharesDBPM(t *testing.T) {
 	for _, tc := range test.TestCases {
 		t.Run(tc.Name, func(t *testing.T) {
-			yes, no := dbpm.DivideUpMarketPoolSharesDBPM(tc.Bets, tc.ProbabilityChanges)
+			yes, no := dbpm.DivideUpMarketPoolSharesDBPM(func() *setup.EconomicConfig { return setup.BuildInitialMarketAppConfig(t, .5, 1, 0, 0) }, tc.Bets, tc.ProbabilityChanges)
```
Comment (ajlacey) [inline backend/tests/dbpm/1_divideupmarketpoolsharesdbpm_test.go:13]: this isn't passing, but does inject a basic default config for these tests, I need to figure out if they're dependent on a specific config type, then I can add it to the test cases. These tests should ultimately be migrated to the dbpm package though.

**Review by ajlacey — COMMENTED** [review]
Added notes to assist in review

**Comment by pwdel** [comment]
@ajlacey Tried to merge from main, there are conflicts. I didn't want to reverse anything that you have done.

**Comment by ajlacey** [comment]
I think I messed up the merge. I have no idea why frontend changes are being show in the PR, so I'm just going to start over at this point...

**Pull request closed without merging**